plugins/Overlay/javascripts/Piwik_Overlay.js

```diff
@@ -26,7 +29,9 @@ var Piwik_Overlay = (function () {
         $location.html('&nbsp;').unbind('mouseenter').unbind('mouseleave');
 
         iframeCurrentPage = currentUrl;
-        iframeDomain = currentUrl.match(/http(s)?:\/\/(www\.)?([^\/]*)/i)[3];
+        var m = currentUrl.match(DOMAIN_PARSE_REGEX);
+        iframeDomain = m[3];
+        iframeOrigin = m[0];
```

Comment:
I presume an attacker cannot set a different iframeOrigin to her or his liking? eg by crafting a url that matches the regex etc... Also `m[0]` and `m[3]` might not be defined for whatever reason... might need to add some check there to make sure they are set...

I'm thinking would it be possible for additional security to only accept origins of configured siteURLs? I think this check is done so far maybe in the twig but not sure if in JS